packages/astro/src/core/build/internal.ts

```diff
@@ -115,8 +112,9 @@ export function* getPageDatasByClientOnlyChunk(
 	const pagesByClientOnly = internals.pagesByClientOnly;
 	if (pagesByClientOnly.size) {
 		for (const [modulePath] of Object.entries(chunk.modules)) {
-			if (pagesByClientOnly.has(modulePath)) {
-				for (const pageData of pagesByClientOnly.get(modulePath)!) {
+			const pathname = `/@fs${prependForwardSlash(modulePath)}`;
+			if (pagesByClientOnly.has(pathname)) {
```

Comment:
module path is now the full module ID, prepending `/@fs` to match the use in `transform()` to match the `trackClientOnlyPageDatas()` use (above)